saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -87,21 +87,39 @@ class Meta:
     @classmethod
     def perform_mutation(cls, _root, info, ids, **data):
         _, pks = resolve_global_ids_to_primary_keys(ids, Product)
-        variants = models.ProductVariant.objects.filter(product__pk__in=pks)
+        product_to_variant = models.ProductVariant.objects.filter(
+            product__pk__in=pks
+        ).values_list("product_id", "id")
+        variants_ids = [variant_id for _, variant_id in product_to_variant]
+
         # get draft order lines for products
         order_line_pks = list(
             order_models.OrderLine.objects.filter(
-                variant__in=variants, order__status=OrderStatus.DRAFT
+                variant_id__in=variants_ids, order__status=OrderStatus.DRAFT
             ).values_list("pk", flat=True)
         )
-
-        response = super().perform_mutation(_root, info, ids, **data)
+        data["product_to_variant"] = product_to_variant
+        response = super().perform_mutation(
+            _root, info, ids, manager=info.context.plugins, **data
+        )
 
         # delete order lines for deleted variants
         order_models.OrderLine.objects.filter(pk__in=order_line_pks).delete()
 
         return response
 
+    @classmethod
+    def bulk_action(cls, queryset, manager, product_to_variant):
+        product_variant_map = defaultdict(list)
+        for product, variant in product_to_variant:
+            product_variant_map[product].append(variant)
+
+        products = [product for product in queryset]
+        queryset.delete()
+        for product in products:
+            variants = product_variant_map.get(product.id)
+            manager.product_deleted(product, variants)
```

Comment:
Here we'll send multiple webhooks in a loop. I think it's fine for now, but in the long run, wouldn't it be better to be able to return a list of items in a single webhook? I'm not sure how it works in other systems. @patrys Do you have any thoughts on this?

Reply:
@d-wysocki  Can you confirm that you fully tested it? I mean - did you subscribe to the webhook product_deleted and checked the payload which we send?

Reply:
@korycins Yes, I tested it with mock server.

Reply:
Could you post here (or in the PR description) the webhook payload which you got?

Reply:
```
b'[{"type": "Product", "id": "UHJvZHVjdDo3Mw==", "variants": [206, 207, 205], "private_metadata": {}, "metadata": {"vatlayer.code": "standard", "vatlayer.description": "standard"}, "name": "Carrot Juice", "description": {"blocks": [{"data": {"text": "Improve your eyesight the natural way with 100% pure, squeezed carrot juice. The sweet, orange nectar of Mother Earth."}, "type": "paragraph"}]}, "updated_at": "2021-01-26T12:06:38.764Z", "charge_taxes": true, "weight": "1.0:kg"}]'
```